### PR TITLE
Update to latest telemetry module

### LINKED ISSUE #132: Please use the latest version of applicationinsights and vscode-extension-telemetry
Hello,

This extension uses `0.21.0` version of applicationInsights which monkey patches the `require` module. This can have unforeseen consequences in the extension host of VS Code. The  `0.0.15` version of `vscode-extension-telemetry` uses an older version of applicationInsights module causing the same issue.

Please update to using the latest version `1.0.6` of applicationinsights and `0.1.0` version of `vscode-extension-telemetry` that has the fix to avoid such monkey patches.

For more details, see https://github.com/Microsoft/ApplicationInsights-node.js/issues/425

Thanks!

### PR DESCRIPTION
**Pull request by StephenWeatherford**
Fixes #132

### CODE REVIEW

**Comment by StephenWeatherford** [comment]
ping

File: package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "Deployment"
   ],
   "repository": {
-    "url": "https://github.com/Microsoft/vscode-azurearmtools.git"
+    "type": "git",
```
Comment (StephenWeatherford) [inline package.json]: Didn't mean to add it to this PR but oh well.  vscode guys say it needs this.

File: src/AzureRMTools.ts [hunk 2]
```diff
@@ -351,11 +351,8 @@ export class AzureRMTools {
      * in a relatively stable state, such as after first opening
      */
     private logFunctionCounts(deploymentTemplate: DeploymentTemplate): void {
-        let me = this;
-        let outputChannelIgnoreForNow: vscode.OutputChannel;
-
         // Don't wait for promise
-        let dummyPromise = callWithTelemetryAndErrorHandling("tle.stats", reporter, outputChannelIgnoreForNow, async function (this: IActionContext) {
+        let dummyPromise = callWithTelemetryAndErrorHandling("tle.stats", async function (this: IActionContext) {
```
Comment (EricJizbaMSFT) [inline src/AzureRMTools.ts]: You'll have to call registerUIExtensionVariables for this to work

File: package.json [hunk 3]
```diff
@@ -90,7 +91,7 @@
     "applicationinsights": "0.21.0",
     "moment": "^2.17.1",
     "open": "^0.0.5",
-    "vscode-azureextensionui": "~0.12.0",
-    "vscode-extension-telemetry": "^0.0.15"
+    "vscode-azureextensionui": "~0.18.0",
+    "vscode-extension-telemetry": "0.0.22"
```
Comment (ramya-rao-a) [inline package.json]: Please update this to 0.1.0 instead. We recently learned of an issue where there are multiple powershell processes being spawned and stay stuck on older Windows versions when using version 1.0.5 of `applicationInsights` module.

This [bug](https://github.com/Microsoft/ApplicationInsights-node.js/issues/438) has been fixed in  version 1.0.6 of `applicationInsights` module which is being used by version 0.1.0 of `vscode-extension-telemtetry`

Therefore, please update the version of `vscode-extension-telemetry` dependency to 0.1.0

**Comment by StephenWeatherford** [comment]
Superseded by #169

**Pull request closed without merging**